### PR TITLE
Support continuous profiling feature

### PR DESCRIPTION
**Pull request by mrproliu**
Features: 
1. Collect process-level-based metrics from continuous profiling.
2. Implement continuous profiling related query(GraphQL) and data collection(gRPC) query.

@lujiajing1126 The Continuous profiling policy is using the `property` type to store into the BanyanDB, please help to recheck. Also, for now, the BanyanDB cannot query `property` with multiple IDs, maybe we can add it. 

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

**Comment by lujiajing1126** [comment]
I suppose property-related features may block due to #10442

File: oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java [hunk 1]
```diff
@@ -108,24 +109,15 @@ protected void apply(StreamQuery query) {
     }
 
     @Override
-    public EBPFProfilingTask queryById(String id) throws IOException {
+    public List<EBPFProfilingTaskRecord> queryByLogicalId(String id) throws IOException {
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java:112]: What does logic ID mean?

**Comment by wu-sheng** [comment]
> I suppose `property`-related features may block due to #10442

Due to this, policy reading should be fail, right?

___
@mrproliu Do we have e2e support for this? I think rover side is ready. We need CLI for testing?

File: oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java (hunk 1 above)
Comment (mrproliu) [reply oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java:112]: logical ID is used on the network profiling, we need one ID to keep the profiling task alive if needed.

**Comment by mrproliu** [comment]
> > I suppose `property`-related features may block due to #10442
> 
> Due to this, policy reading should be fail, right?
> 
> @mrproliu Do we have e2e support for this? I think rover side is ready. We need CLI for testing?

I'm working on adapting the query protocol to the CLI, then I will create the E2E for testing this feature in difference storage.

File: oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java (hunk 1 above)
Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java:112]: I mean, is it task ID or something? From a naming perspective, logic ID looks like a general name, I want it relative with profiling task.

Comment (mrproliu) [reply oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java:112]: Oh, it's task ID. I just let the DAO return the original task storage data, not the query data. Then, the service decides how to use the stored data(reduce the logical code in the DAO).

Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java:112]: OK, then rename this to `getTaskRecord`?

Comment (mrproliu) [reply oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBEBPFProfilingTaskDAO.java]: Ok, renamed.

**Comment by wu-sheng** [comment]
Notice, due to https://github.com/apache/skywalking/issues/10366, MySQL DAO could be impacted. 
Let's hold this PR for a while, I think @kezhenxu94 will make https://github.com/apache/skywalking/issues/10366 passed in this week.

**Comment by wu-sheng** [comment]
Also, I would remove this from 9.4.0 release. Because it is impossible we could provide UI for this in time. WDYT? @mrproliu

**Comment by mrproliu** [comment]
> Also, I would remove this from 9.4.0 release. Because it is impossible we could provide UI for this in time. WDYT? @mrproliu

Sure.

**Comment by hanahmily** [comment]
> Also, for now, the BanyanDB cannot query property with multiple IDs

The API provides listing multi-ids in a single request. @lujiajing1126 Java client should support it accordingly.

**Comment by hanahmily** [comment]
@mrproliu @wu-sheng Could we set up a meeting to discuss how to store profiling data to BanyanDB in an appropriate way? The BanyanDB project will kick off the design process of distribution. We should align some issues before that.

**Comment by wu-sheng** [comment]
Sure, ping me, I am flexible these days.

**Comment by mrproliu** [comment]
Sure.

File: oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/handler/ContinuousProfilingServiceHandler.java [hunk 2]
```diff
@@ -64,10 +71,14 @@ public class ContinuousProfilingServiceHandler extends ContinuousProfilingServic
 
     private IContinuousProfilingPolicyDAO policyDAO;
     private final CommandService commandService;
+    private final Cache<String, PolicyWrapper> policyCache;
 
-    public ContinuousProfilingServiceHandler(ModuleManager moduleManager) {
+    public ContinuousProfilingServiceHandler(ModuleManager moduleManager, EBPFReceiverModuleConfig config) {
         this.policyDAO = moduleManager.find(StorageModule.NAME).provider().getService(IContinuousProfilingPolicyDAO.class);
         this.commandService = moduleManager.find(CoreModule.NAME).provider().getService(CommandService.class);
+        this.policyCache = CacheBuilder.newBuilder()
```
Comment (wu-sheng) [inline oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/handler/ContinuousProfilingServiceHandler.java:79]: How about using `.build(new CacheLoader`?

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class JDBCContinuousProfilingPolicyDAO extends JDBCSQLExecutor implements IContinuousProfilingPolicyDAO {
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java:45]: @kezhenxu94 Could you recheck this new JDBC Dao? As several new implementations are just added, I want to make sure the APIs are used correctly.

File: oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBContinuousProfilingPolicyDAO.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            .addTag(TagAndValue.newStringTag(ContinuousProfilingPolicy.UUID, policy.getUuid()))
+            .addTag(TagAndValue.newStringTag(ContinuousProfilingPolicy.CONFIGURATION_JSON, policy.getConfigurationJson()))
+            .build();
+    }
+
+    @Override
+    public List<ContinuousProfilingPolicy> queryPolicies(List<String> serviceIdList) throws IOException {
+        return serviceIdList.stream().map(s -> {
+            try {
+                return getClient().queryProperty(GROUP, ContinuousProfilingPolicy.INDEX_NAME, s);
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/stream/BanyanDBContinuousProfilingPolicyDAO.java:61]: @hanahmily @lujiajing1126 How about the plans of BanyanDB for bulk property query? Should we add an issue and TODO to track this?

File: oap-server/server-starter/src/main/resources/application.yml [hunk 5]
```diff
@@ -527,6 +527,8 @@ receiver-event:
 receiver-ebpf:
   selector: ${SW_RECEIVER_EBPF:default}
   default:
+    # The continuous profiling policy cache time, Unit is s.
+    continuousPolicyCacheTimeout: ${SW_CONTINUOUS_POLICY_CACHE_TIMEOUT:60}
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/application.yml:531]: New configuration should be added in the doc.

```suggestion
    # The continuous profiling policy cache time, Unit is second.
    continuousPolicyCacheTimeout: ${SW_CONTINUOUS_POLICY_CACHE_TIMEOUT:60}
```

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java (hunk 3 above)
Comment (mrproliu) [reply oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java:45]: Also, the `JDBCEBPFProfilingTaskDAO` has been updated, it should be re-checked too.

**Comment by mrproliu** [comment]
> profiling.md doc should be updated about continuous profiling, right?

Yes, I will update it.

**Review by wu-sheng — APPROVED** [review]
1. Protocol has been reviewed several times before. 
2. The rover has been tested through a 100% load test with a 10% continuous profiling policy.
3. Docs are mostly TODO.

Generally good to me.

File: oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/EBPFReceiverModuleConfig.java [hunk 6]
```diff
@@ -26,7 +26,7 @@
 public class EBPFReceiverModuleConfig extends ModuleConfig {
 
     /**
-     * The continuous profiling policy cache time, Unit is s. Default value is 60s.
+     * The continuous profiling policy cache time, Unit is second. Default value is 60 second.
```
Comment (wu-sheng) [inline oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/EBPFReceiverModuleConfig.java:29]: 60 seconds.

**Review by wu-sheng — DISMISSED** [review]

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    @Override
+    public List<ContinuousProfilingPolicy> queryPolicies(List<String> serviceIdList) throws IOException {
+        final var tables = tableHelper.getTablesWithinTTL(ContinuousProfilingPolicy.INDEX_NAME);
+        final StringBuilder sql = new StringBuilder();
+        List<Object> condition = new ArrayList<>();
+        condition.add(ContinuousProfilingPolicy.INDEX_NAME);
+        condition.addAll(serviceIdList);
+
+        for (String table : tables) {
+            sql.append("select * from ").append(table).append(" where ")
```
Comment (kezhenxu94) [inline oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java:78]: As `sql` is used inside a loop, you need to move the declaration into the loop, or `clear` the `sql` instance.

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+                .append(JDBCTableInstaller.TABLE_COLUMN).append(" = ?")
+                .append(" and ").append(ContinuousProfilingPolicy.SERVICE_ID)
+                .append(" in (").append(Joiner.on(",").join(serviceIdList.stream().map(s -> "?").collect(Collectors.toList())))
+                .append(" )");
+
+            return jdbcClient.executeQuery(sql.toString(), this::buildPolicies, condition.toArray(new Object[0]));
```
Comment (kezhenxu94) [inline oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java:84]: Maybe you don't want to return here (inside a for loop)?

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        List<Object> condition = new ArrayList<>();
+        condition.add(ContinuousProfilingPolicy.INDEX_NAME);
+        condition.addAll(serviceIdList);
+
+        for (String table : tables) {
+            sql.append("select * from ").append(table).append(" where ")
+                .append(JDBCTableInstaller.TABLE_COLUMN).append(" = ?")
+                .append(" and ").append(ContinuousProfilingPolicy.SERVICE_ID)
+                .append(" in (").append(Joiner.on(",").join(serviceIdList.stream().map(s -> "?").collect(Collectors.toList())))
+                .append(" )");
```
Comment (kezhenxu94) [inline oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java:82]: nit

```suggestion
                .append(" in ").append(serviceIdList.stream().map(s -> "?").collect(Collectors.joining(",", "(", ")")));
```

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java (hunk 7 above)
Comment (mrproliu) [reply oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java]: fixed, please re-check.

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+        StringBuilder whereQuery = new StringBuilder()
+            .append(" where ")
+            .append(JDBCTableInstaller.TABLE_COLUMN).append(" = ?")
+            .append(" and ").append(ContinuousProfilingPolicy.SERVICE_ID)
+            .append(" in ").append(serviceIdList.stream().map(s -> "?").collect(Collectors.joining(",", "(", ")")));
+        for (String table : tables) {
+            return jdbcClient.executeQuery("select * from " + table + whereQuery, this::buildPolicies, condition.toArray(new Object[0]));
+        }
+
+        return Collections.emptyList();
```
Comment (kezhenxu94) [inline oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java:84]: This is not fixed. You might want to change to 

```suggestion
        final var results = new ArrayList<ContinuousProfilingPolicy>();
        for (String table : tables) {
            results.addAll(jdbcClient.executeQuery("select * from " + table + whereQuery, this::buildPolicies, condition.toArray(new Object[0])));
        }
        return results;
```

Comment (mrproliu) [reply oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java]: Ohh, I forget to change. been updated, please re-check. :)

**Pull request merged**